src/components/searchBar.tsx

```diff
@@ -0,0 +1,97 @@
+'use client'
+import {
+    Form,
+    FormControl,
+    FormDescription,
+    FormField,
+    FormItem,
+} from '@/components/ui/form'
+import { Input } from '@/components/ui/input'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { Loader2, Search } from 'lucide-react'
+import { useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import toast from 'react-hot-toast'
+import { z } from 'zod'
+import { Button } from './ui/button'
+import { useRouter } from 'next/navigation'
+
+const searchSchema = z.object({
+    searchTerm: z.string().min(1),
+})
+type searchSchema = z.infer<typeof searchSchema>
+export default function SearchBar({ prevSearch }: { prevSearch?: string }) {
```

Comment:
I don't like the naming of prop "prevSearch", I would call it something like `initialSearchTerm` instead. There is no previous search if I'm linked to the page